pkg/osquery/tables/dataflattentable/tables_test.go

```diff
@@ -78,16 +82,76 @@ func TestDataFlattenTable_Animals(t *testing.T) {
 
 }
 
-func TestDataFlattenIniTable(t *testing.T) {
+func TestDataFlattenTables(t *testing.T) {
 	t.Parallel()
 
-	testTable := Table{dataFunc: dataflatten.IniFile}
-	mockQC := tablehelpers.MockQueryContext(map[string][]string{
-		"path": []string{path.Join("testdata", "secdata.ini")},
-	})
+	logger := log.NewNopLogger()
+
+	var tests = []struct {
+		testTables   map[string]Table
+		testFile     string
+		queries      []string
+		expectedRows int
+		expectNoData bool
+	}{
+		// xml
+		{
+			testTables:   map[string]Table{"xml": Table{logger: logger, dataFunc: dataflatten.XmlFile}},
+			testFile:     path.Join("testdata", "simple.xml"),
+			expectedRows: 6,
+		},
+		{
+			testTables:   map[string]Table{"xml": Table{logger: logger, dataFunc: dataflatten.XmlFile}},
+			testFile:     path.Join("testdata", "simple.xml"),
+			queries:      []string{"simple/Items"},
+			expectedRows: 3,
+		},
+		{
+			testTables:   map[string]Table{"xml": Table{logger: logger, dataFunc: dataflatten.XmlFile}},
+			testFile:     path.Join("testdata", "simple.xml"),
+			queries:      []string{"this/does/not/exist"},
+			expectNoData: true,
+		},
+
+		// ini
+		{
+			testTables:   map[string]Table{"ini": Table{logger: logger, dataFunc: dataflatten.IniFile}},
+			testFile:     path.Join("testdata", "secdata.ini"),
+			expectedRows: 87,
+		},
+		{
+			testTables:   map[string]Table{"ini": Table{logger: logger, dataFunc: dataflatten.IniFile}},
+			testFile:     path.Join("testdata", "secdata.ini"),
+			queries:      []string{"Registry Values"},
+			expectedRows: 59,
+		},
+		{
+			testTables:   map[string]Table{"ini": Table{logger: logger, dataFunc: dataflatten.IniFile}},
+			testFile:     path.Join("testdata", "secdata.ini"),
+			queries:      []string{"this/does/not/exist"},
+			expectNoData: true,
```

Comment:
Nit: could this just be `expectedRows: 0`, and remove the conditional on line 147? (seemed to work as expected locally)

Reply:
It _could_ but the test cases are kinda a mess. Some use `expectedRows` and some use `minRows` and `noData`.  I'm going to leave this as tech debt for a future pass